src/test/java/TasksServletTest.java

```diff
@@ -116,33 +91,23 @@
           TASK_COMPLETED_YESTERDAY,
           TASK_COMPLETED_TODAY);
 
+  @Override
   @Before
-  public void setUp() throws IOException, GeneralSecurityException {
-    authenticationVerifier = Mockito.mock(AuthenticationVerifier.class);
+  public void setUp() throws Exception {
+    super.setUp();
+
     tasksClientFactory = Mockito.mock(TasksClientFactory.class);
     tasksClient = Mockito.mock(TasksClient.class);
     servlet = new TasksServlet(authenticationVerifier, tasksClientFactory);
 
     Mockito.when(tasksClientFactory.getTasksClient(Mockito.any())).thenReturn(tasksClient);
-    // Authentication will always pass
-    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
-        .thenReturn(AUTHENTICATION_VERIFIED);
-
-    request = Mockito.mock(HttpServletRequest.class);
-    response = Mockito.mock(HttpServletResponse.class);
-    Mockito.when(request.getCookies()).thenReturn(validCookies);
-
-    // Writer used in get/post requests to capture HTTP response values
-    stringWriter = new StringWriter();
-    printWriter = new PrintWriter(stringWriter);
-    Mockito.when(response.getWriter()).thenReturn(printWriter);
   }
 
   @Test
-  public void noTaskLists() throws IOException, ServletException {
+  public void noTaskLists() throws Exception {
     Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASKLISTS);
     servlet.doGet(request, response);
-    printWriter.flush();
+
```

Comment:
Just curious because I might change this for my code as well - is `printWriter.flush()` not needed?

Reply:
Thanks for pointing this out - You can actually create a printWriter with autoFlush (which will flush the stream every time `printWriter.println()` is called. I just put this into the `HttpServletResponseFake` as I initially intended to (it wasn't there when you reviewed).

Reply:
Oh that's neat, thanks!